### PR TITLE
forward inline information

### PR DESCRIPTION
**Pull request by lievenhey**
This should allow hotspot to show inline information. Currently it doesn't work for every symbol.
If you see something like this:
grafik
in the flamegraph, it doesn't work. The parent frame will have the inline information, but not top child.

### CODE REVIEW

**Comment by lievenhey** [comment]
see https://github.com/KDAB/hotspot/pull/568

File: app/perfunwind.cpp [hunk 1]
```diff
@@ -506,7 +506,7 @@ QDataStream &operator<<(QDataStream &stream, const PerfUnwind::Location &locatio
 
 QDataStream &operator<<(QDataStream &stream, const PerfUnwind::Symbol &symbol)
 {
-    return stream << symbol.name << symbol.binary << symbol.path << symbol.isKernel << symbol.relAddr << symbol.size << symbol.actualPath;
+    return stream << symbol.name << symbol.binary << symbol.path << symbol.isKernel << symbol.isInline << symbol.relAddr << symbol.size << symbol.actualPath;
```
Comment (milianw) [inline app/perfunwind.cpp]: please always add the new data at the end such that the format stays somewhat compatible to before

File: app/perfsymboltable.cpp [hunk 2]
```diff
@@ -282,10 +282,10 @@ int PerfSymbolTable::parseDie(CuDieRangeMapping *cudie, Dwarf_Die *top, quint64
         location.parentLocationId = parentLocationId;
 
         int callLocationId = m_unwind->resolveLocation(location);
-        return insertSubprogram(cudie, top, entry, offset, size, relAddr, binaryId, binaryPathId, actualPathId, callLocationId, isKernel);
+        return insertSubprogram(cudie, top, entry, offset, size, relAddr, binaryId, binaryPathId, actualPathId, callLocationId, isKernel, true);
```
Comment (milianw) [inline app/perfsymboltable.cpp:285]: are you sure this is the right place to make the inline vs. non-inline decision? see `PerfSymbolTable::parseDwarf` which gets a clear list of inlined DIEs and the subprogram die - I think that's the level where this should be handled, no?

**Review by milianw — CHANGES_REQUESTED** [review]
don't we have tests for this part here too that should be extended then? e.g. in TestPerfData and PerfParserTestClient you should also get this data and verify that this works as you expect it to

File: app/perfsymboltable.cpp (hunk 2 above)
Comment (lievenhey) [reply app/perfsymboltable.cpp:285]: `PerfSymbolTable::parseDwarf` calls `PerfSymbolTable::handleDie` so I think its fine.

**Comment by milianw** [comment]
> don't we have tests for this part here too that should be extended then? e.g. in TestPerfData and PerfParserTestClient you should also get this data and verify that this works as you expect it to

this isn't done yet

**Review by milianw — APPROVED** [review]

**Comment by lievenhey** [comment]
now it is done. Somewhat crude but working.

**Pull request merged**